### ISSUE TITLE
CEF lagging when your VRAM is too low

### ISSUE THREAD

**Issue by martin839** [altmp/altv-issues#612]
Hello,

today I (and many other players) have noticed, that if you have only like 4GB of VRAM, your CEF is lagging sometimes (like 1-3 seconds delay).

Regards

**Comment by LeonMrBonnie** [comment]
I don't think this has anything to do with VRAM. It is probably the same bug VnX SolidSnake had. (https://github.com/altmp/altv-issues/issues/588)

**Comment by gamelaster** [comment]
Hello @martin839 ,

please, give us more information about your GPU and your graphic configuration in the game.

Thanks.

**Comment by martin839** [comment]
Hi @gamelaster ,

I have a GTX 780 with 3GB of VRAM. It's enough to play on high, but when I put my textures und high+, I have stable FPS but the CEF is lagging/delayed. When I put that down to normal, it's fine again.

**Comment by gamelaster** [comment]
Well, from the minimal / recommended requirements for GTA5, at least 2 GB VRAM is required, so I think that your GPU just can't handle GTA + CEF at same time in good speed, so only solution is just to use lower game configuration.

Regards

**Comment by martin839** [comment]
Isn't it possible to optimize it or make it more cpu heavy than gpu?

**Comment by LeonMrBonnie** [comment]
CEF is not made by the alt:V devs. So no, there is not really anything they can do.

**Comment by gamelaster** [comment]
@martin839 
> Isn't it possible to optimize it

Sadly, at the moment, we optimized all the things we are able in side of CEF implementation, other things are under the CEF and Chromium hood.

> make it more cpu heavy than gpu

This is actually possible, but you really don't want it, software rendering is really CPU expensive, that's why we using GPU acceleration.

Only solution is to switch to another engine, but actually, there isn't any better solution, and everyone is already very used to CEF and making UIs in HTML, CSS and JS, so it's very unlikely to change to another engine in near time. (Even we tried, sadly, no better alternatives are yet available).
